lib/std/x/os/socket_windows.zig

```diff
@@ -0,0 +1,392 @@
+                .WSAEDISCON => error.ConnectionClosedByPeer,
+                .WSAEFAULT => error.BadBuffer,
+                .WSAEINPROGRESS,
+                .WSAEWOULDBLOCK,
+                .WSA_IO_PENDING,
+                .WSAETIMEDOUT,
+                => error.WouldBlock,
+                .WSAEINTR => error.Cancelled,
+                .WSAEINVAL => error.SocketNotBound,
+                .WSAEMSGSIZE => error.MessageTooLarge,
+                .WSAENETDOWN => error.NetworkSubsystemFailed,
+                .WSAENETRESET => error.NetworkReset,
+                .WSAENOTCONN => error.SocketNotConnected,
+                .WSAENOTSOCK => error.FileDescriptorNotASocket,
+                .WSAEOPNOTSUPP => error.OperationNotSupported,
+                .WSAESHUTDOWN => error.AlreadyShutdown,
+                .WSA_OPERATION_ABORTED => error.OperationAborted,
+                else => |err| windows.unexpectedWSAError(err),
+            };
+        }
+
+        return @intCast(usize, rc);
+    }
+
+    /// Write a buffer of data provided to the socket with a set of flags specified.
+    /// It returns the number of bytes that are written to the socket.
+    pub fn write(self: Socket, buf: []const u8, flags: u32) !usize {
+        var bufs = &[_]ws2_32.WSABUF{.{ .len = @intCast(u32, buf.len), .buf = buf.ptr }};
+        var flags_ = flags;
+
+        const rc = ws2_32.WSASend(self.fd, bufs, 1, null, &flags_, null, null);
+        if (rc == ws2_32.SOCKET_ERROR) {
+            return switch (ws2_32.WSAGetLastError()) {
+                .WSAECONNABORTED => error.ConnectionAborted,
+                .WSAECONNRESET => error.ConnectionResetByPeer,
+                .WSAEFAULT => error.BadBuffer,
+                .WSAEINPROGRESS,
+                .WSAEWOULDBLOCK,
+                .WSA_IO_PENDING,
+                .WSAETIMEDOUT,
+                => error.WouldBlock,
+                .WSAEINTR => error.Cancelled,
+                .WSAEINVAL => error.SocketNotBound,
+                .WSAEMSGSIZE => error.MessageTooLarge,
+                .WSAENETDOWN => error.NetworkSubsystemFailed,
+                .WSAENETRESET => error.NetworkReset,
+                .WSAENOBUFS => error.BufferDeadlock,
+                .WSAENOTCONN => error.SocketNotConnected,
+                .WSAENOTSOCK => error.FileDescriptorNotASocket,
+                .WSAEOPNOTSUPP => error.OperationNotSupported,
+                .WSAESHUTDOWN => error.AlreadyShutdown,
+                .WSA_OPERATION_ABORTED => error.OperationAborted,
+                else => |err| windows.unexpectedWSAError(err),
+            };
+        }
+
+        return @intCast(usize, rc);
+    }
+
+    /// Writes multiple I/O vectors with a prepended message header to the socket
+    /// with a set of flags specified. It returns the number of bytes that are
+    /// written to the socket.
+    pub fn writeVectorized(self: Socket, msg: os.msghdr_const, flags: u32) !usize {
+        return error.NotImplemented;
```

Comment:
Just don't include the function for now?

Reply:
Looking to have it just be a stub for now, I want to address it in the next PR :smile:.

Reply:
Implemented `{read, write}Vectorized()` for both Windows and POSIX in a new commit. More work could be done to make the abstraction much better, though that will be work for subsequent PR's.